File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.cpp
class Solution {
public:
    vector<bool> checkArithmeticSubarrays(vector<int>& nums, vector<int>& l, vector<int>& r) {
        vector<bool> res;
        
        for(int i = 0;i<l.size();i++){
            vector<int> a(begin(nums)+l[i],begin(nums)+r[i]+1);
            res.push_back(check(a));
        }
        return res;
    }
    
    bool check(vector<int>& a){
        sort(a.begin(),a.end());
        
        int diff = a[1] - a[0];
        
        for(int i =2;i<a.size();i++){
            if(a[i] - a[i-1] != diff){
                return false;
            }
        }
        
        return true;
    }
};

File: 1630-arithmetic-subarrays/1630-arithmetic-subarrays.java
class Solution {
    public List<Boolean> checkArithmeticSubarrays(int[] nums, int[] l, int[] r) {
        List<Boolean> res = new ArrayList<>();
        for(int i = 0;i<l.length;i++){
            int[] a = new int[r[i] - l[i]+1];
            for(int j = 0;j<a.length;j++){
                a[j] = nums[l[i]+j];
            }
            res.add(check(a));
        }
        return res;
    }
    
    public Boolean check(int[] a){
        Arrays.sort(a);
        int diff = a[1] - a[0];
        
        for(int i = 2;i<a.length;i++){
            if(a[i] - a[i-1] != diff){
                return false;
            }
        }
        return true;
    }
}